### PR TITLE
[#180] Document print behavior in python rules (main)

### PR DESCRIPTION
**Pull request by alanking**
In service of #180

### CODE REVIEW

**Comment by korydraughn** [comment]
Probably worth mentioning the NREP is required to be configured before calling `writeLine`.

**Comment by trel** [comment]
oh, is that true?   not a system microservice?

**Comment by korydraughn** [comment]
There's no such thing as a system MSI AFAIK.

I'm pretty sure `writeLine` is part of native rule engine plugin. If not, then that's news to me.

**Comment by trel** [comment]
grep for `smsi`

**Comment by trel** [comment]
with only PREP and default_cpp installed...

core.py
```
def acPostProcForPut(rule_args, callback, rei):
    callback.writeLine('serverLog', 'callback.writeLine without NREP!')
```

```
 {"log_category":"legacy","log_level":"info","log_message":"writeLine: inString = callback.writeLine without NREP!\n","request_api_name":"DATA_OBJ_PUT_AN","request_api_number":606,"request_api_version":"d","request_client_user":"rods","request_host":"127.0.0.1","request_proxy_user":"rods","request_release_version":"rods4.3.1","server_host":"xxx.xxx.xxx.xxx","server_pid":3942678,"server_timestamp":"2023-11-21T23:00:58.813Z","server_type":"agent","server_zone":"tempZone"}
```

**Comment by korydraughn** [comment]
Looks like we're all set then.

File: README.md [hunk 1]
```diff
@@ -420,3 +420,21 @@ The following arguments are strictly optional and may be used either in `Query`'
...
   * *case-sensitive* is normally `True`. If it is set to `False`, the condition string will be uppercased, and the query will be executed without regard to case.  This allows for more permissive matching on the names of resources, collections, data objects, etc.
   * *options* is a bitmask of extra options, and defaults to a value of 0. `genquery.Option.NO_DISTINCT` is one such extra option, as is RETURN_TOTAL_ROW_COUNT (although in the latter case, using the `Query` object's `row_count` method should be preferred.)
+
+## Questions and Answers
+
+### What happened to my `print` output?
+
+In iRODS server versions 4.3.0 and later, all standard streams are redirected to `/dev/null` in the server. This means that any data sent to `stdout` such as via a `print` statement in a rule run inside the server will be discarded.
+
+Here is some example code which would print a message to the `rodsLog` in servers from the 4.2.x series and earlier when run from a PREP rule:
```
Comment (korydraughn) [inline README.md:430]: `PREP` -> `Python Rule Engine Plugin (PREP)`?

Comment (alanking) [reply README.md]: I just got rid of the PREP usage because it is not referenced again.

**Review by korydraughn — APPROVED** [review]
Squash/Pound if you're happy with it.

**Pull request merged**